src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -1650,9 +1650,21 @@ public void testAllStyleRules() throws Exception {
                             fileName, ruleName);
                 }
 
-                validateStyleModules(XmlUtil.findChildElementsByTag(columns.get(2), "a"),
-                        XmlUtil.findChildElementsByTag(columns.get(3), "a"), styleChecks, styleName,
-                        ruleName);
+                final Iterator<Node> itrChecks =
+                        XmlUtil.findChildElementsByTag(columns.get(2), "a").iterator();
+                final Iterator<Node> itrConfigs =
+                        XmlUtil.findChildElementsByTag(columns.get(3), "a").iterator();
+
+                if ("google".equals(styleName)) {
+                    final Iterator<Node> itrSamples =
+                            XmlUtil.findChildElementsByTag(columns.get(4), "a").iterator();
+                    validateStyleModulesGoogle(itrChecks, itrConfigs, itrSamples,
```

Comment:
I would name these more generic. Originally this was chapter wise testing and other testing. Yes, google is the only one using this but who is to say Sun won't follow suite? It certainly doesn't make sense we have 2 separate styles. IMO, we should be uniform.

Reply:
I have reverted all these changes in the latest push. Now there's much less extra code, mostly existing code is modified